.github/workflows/build.yml

```diff
@@ -0,0 +1,115 @@
+        cp ./Dockerfile.tmplate "$DOCKERFILE"
+        cat << EOF >> "$DOCKERFILE"
+        CMD ["${{ fromJSON(toJSON(matrix)).target }}"]
+        EOF
+      context: .
+      buildArgs: |
+        BUILD=${{ fromJSON(toJSON(matrix)).target }}
+        VERSION=git-${{ needs.prepare.outputs.git-rev }}
+        ASSET_DIR=${{ fromJSON(toJSON(matrix)).folder }}/assets
+      dockerfile: ${{ fromJSON(toJSON(matrix)).folder }}/${{ fromJSON(toJSON(matrix)).target }}.Dockerfile
+      imageName: ${{ fromJSON(toJSON(matrix)).target }}
+      platforms: linux/amd64
+      push: ${{ github.ref == 'refs/heads/main' }}
+      tags: latest,git-${{ needs.prepare.outputs.git-rev }}
+      registryOverride: 862640294325.dkr.ecr.ap-southeast-2.amazonaws.com
+      aws-region: ap-southeast-2
+      aws-role-arn-to-assume: arn:aws:iam::862640294325:role/github-actions-geonet-ecr-push
+      aws-role-duration-seconds: "3600"
+  go-build:
+    if: ${{ contains(fromJSON('["workflow_call", "push", "pull_request"]'), github.event_name) && startsWith(github.repository, 'GeoNet/') != false }}
+    uses: GeoNet/Actions/.github/workflows/reusable-go-build-smoke-test.yml@main
+    with:
+      paths: ${{ inputs.paths }}
+  gofmt:
+    if: ${{ contains(fromJSON('["workflow_call", "push", "pull_request"]'), github.event_name) && startsWith(github.repository, 'GeoNet/') != false }}
+    uses: GeoNet/Actions/.github/workflows/reusable-gofmt.yml@main
+  golangci-lint:
+    if: ${{ contains(fromJSON('["workflow_call", "push", "pull_request"]'), github.event_name) && startsWith(github.repository, 'GeoNet/') != false }}
+    uses: GeoNet/Actions/.github/workflows/reusable-golangci-lint.yml@main
+  go-vet:
+    if: ${{ contains(fromJSON('["workflow_call", "push", "pull_request"]'), github.event_name) && startsWith(github.repository, 'GeoNet/') != false }}
+    uses: GeoNet/Actions/.github/workflows/reusable-go-vet.yml@main
+  go-test:
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: ap-southeast-2
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version-file: go.mod
+          cache-dependency-path: go.sum
+          check-latest: true
+      - name: setup
+        run: |
+          sudo apt-get -yq update
+          sudo apt-get install -y xsltproc
+          docker \
+          run -d \
+          -p 5432:5432 \
+          -e POSTGRES_PASSWORD=test \
+          -e POSTGRES_USER=fdsn_w \
+          -e POSTGRES_DB=fdsn \
+          --name postgres \
+          docker.io/postgis/postgis:15-3.3-alpine
+          echo "Waiting until Postgres is ready..."
+          until nc -zv -w 1 127.0.0.1 5432; do
+          sleep 1s
+          done
+          sleep 5s
+          docker logs postgres
+          echo "Postgres is ready"
+          psql postgresql://fdsn_w:test@127.0.0.1/fdsn --file=./etc/ddl/drop-create.ddl
```

Comment:
The initdb.sh doesn't seem to be compatible with how Postgres in a container works; e.g dropdb and no password in psql command.
Is it OK to have code change on it to make it compatible?

Reply:
Password is provided by exporting `PGPASSWORD` at L7.

(question: from where do you get the `psql` and `dropdb` command line tool? I can't find it from your code)

If the script can't fit for both local developers and actions, then I would suggest we create a seperate initdb.sh (say, `initdb-workflow.sh` or another name you think would fits better.) This happens in haz: https://github.com/GeoNet/haz/tree/main/etc/scripts

Reply:
In my experience with `psql`, I haven't had success with using the `PGPASSWORD` env; this could've changed in some recent versions but I've only had success with either a URI (i.e: `postgresql://USER:PASSWORD@HOST:PORT/DATABASE`) or with a .pgpass file.

Since Postgres is running in a container, `dropdb` will only be available inside the DB.
I'm not sure it's reasonable to drop the database for a freshly provisioned database, even with the same name and such.

Does it need to be a script or can it be full contained in the workflow?
My thought is, since it's for automation it might like to be in the workflow to be tidy.